packages/core/src/lib/make-theme.mts

```diff
@@ -1,10 +1,40 @@
 import type { Prettify, PartialDeep } from '@inquirer/type';
 import { defaultTheme, type Theme } from './theme.mjs';
 
+function isPlainObject(value: unknown): value is object {
+  if (typeof value !== 'object' || value === null) return false;
+
+  let proto = value;
+  while (Object.getPrototypeOf(proto) !== null) {
+    proto = Object.getPrototypeOf(proto);
+  }
+
+  return Object.getPrototypeOf(value) === proto;
+}
+
+function deepMerge<T extends object>(...objects: Partial<T>[]): T {
+  const output: { [key: string]: unknown } = {};
+
+  for (const obj of objects) {
+    for (const [key, value] of Object.entries(obj)) {
+      const prevValue = output[key];
+
+      output[key] =
+        isPlainObject(prevValue) && isPlainObject(value)
+          ? deepMerge(prevValue, value)
+          : value;
+    }
+  }
+
+  return output as T;
```

Comment:
Those `as` uses aren't safe since they force a type. Ideally we could have TS type check without forcing the types.